src/sentry/integrations/slack/webhooks/action.py

```diff
@@ -382,6 +427,53 @@ def build_archive_modal_payload(self, callback_id):
             "callback_id": callback_id,
         }
 
+    def build_archive_modal_payload(self, callback_id) -> View:
```

Comment:
Let's look for ways to reduce code duplicated from `build_resolve_modal_payload`. Maybe we can extract a few helper functions.